utils/bin/indexer/src/main.rs

```diff
@@ -0,0 +1,100 @@
+use std::{cmp::min, collections::HashMap};
+
+use alloy::{
+    primitives::{Address, Bytes},
+    providers::{Provider, ProviderBuilder},
+    rpc::types::Filter,
+    sol,
+    sol_types::{SolEvent, SolValue},
+};
+
+// Codegen from ABI file to interact with the contract.
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    ComposableCoW,
+    "../../abi/ComposableCoW.json"
+);
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    const WINDOW_SIZE: u64 = 10000;
+    const MAINNET_DEPLOYMENT_GENESIS: u64 = 19277205;
+    const MAINNET_CONSTANT_PRODUCT_HANDLER: &str = "0x34323B933096534e43958F6c7Bf44F2Bb59424DA";
+    const GNOSIS_DEPLOYMENT_GENESIS: u64 = 32478660;
+    const GNOSIS_CONSTANT_PRODUCT_HANDLER: &str = "0xB148F40fff05b5CE6B22752cf8E454B556f7a851";
+
+    let rpc_url = match std::env::var("RPC_URL") {
+        Ok(url) => match url.parse() {
+            Ok(url) => url,
+            Err(_) => {
+                eprintln!("Invalid URL: {}", url);
+                std::process::exit(1);
+            }
+        },
+        Err(_) => {
+            eprintln!("Environment variable `RPC_URL` is not set");
+            eprintln!("Usage: RPC_URL=<URL> indexer");
+            std::process::exit(1);
+        }
+    };
+
+    let provider = ProviderBuilder::new().on_http(rpc_url);
+
+    // Determine the chain and then set the constant product handler and genesis block
+    let chain_id = provider.get_chain_id().await?;
```

Comment:
Maybe also log the `chain_id` here. There have been instances where services stacks were started with an URL for a different chain than expected.

Reply:
Done